src/RestWrite.js

```diff
@@ -464,15 +469,21 @@ RestWrite.prototype.handleFollowup = function() {
         }
     };
     delete this.storage['clearSessions'];
-    this.config.database.destroy('_Session', sessionQuery)
+    return this.config.database.destroy('_Session', sessionQuery)
+    .then(this.handleFollowup.bind(this));
+  }
+  
+  if (this.storage && this.storage['generateNewSession']) {
+    delete this.storage['generateNewSession'];
+    return this.createSessionToken()
```

Comment:
That seem to work correctly as is, as we call the `handleFollowup` after. 

I needed to refactor as synchronous methods so destroy don't race with generation